layer/windows_writer_test.go

```diff
@@ -0,0 +1,229 @@
+
+				f.Seek(0, 0)
+				tr := tar.NewReader(f)
+
+				th, _ := tr.Next()
+				h.AssertEq(t, th.Name, "Files")
+				h.AssertEq(t, th.Typeflag, byte(tar.TypeDir))
+
+				th, _ = tr.Next()
+				h.AssertEq(t, th.Name, "Hives")
+				h.AssertEq(t, th.Typeflag, byte(tar.TypeDir))
+
+				th, _ = tr.Next()
+				h.AssertEq(t, th.Name, "Files/cnb")
+				h.AssertEq(t, th.Typeflag, byte(tar.TypeDir))
+
+				th, _ = tr.Next()
+				h.AssertEq(t, th.Name, "Files/cnb/lifecycle")
+				h.AssertEq(t, th.Typeflag, byte(tar.TypeDir))
+
+				th, _ = tr.Next()
+				h.AssertEq(t, th.Name, "Files/cnb/lifecycle/first-file")
+				h.AssertEq(t, th.Typeflag, byte(tar.TypeReg))
+
+				th, _ = tr.Next()
+				h.AssertEq(t, th.Name, "Files/cnb/sibling-dir")
+				h.AssertEq(t, th.Typeflag, byte(tar.TypeDir))
+
+				_, err = tr.Next()
+				h.AssertError(t, err, "EOF")
+			})
+		})
+
+		when("header path begins without slash or dot", func() {
+			it("writes entries", func() {
+				var err error
+
+				f, err := ioutil.TempFile("", "windows-writer.tar")
+				h.AssertNil(t, err)
+				defer func() { f.Close(); os.Remove(f.Name()) }()
+
+				lw := layer.NewWindowsWriter(f)
+
+				h.AssertNil(t, lw.WriteHeader(&tar.Header{
+					Name:     "cnb/my-file",
+					Typeflag: tar.TypeReg,
+				}))
+
+				h.AssertNil(t, lw.Close())
+
+				f.Seek(0, 0)
+				tr := tar.NewReader(f)
+
+				th, _ := tr.Next()
+				h.AssertEq(t, th.Name, "Files")
+				h.AssertEq(t, th.Typeflag, byte(tar.TypeDir))
+
+				th, _ = tr.Next()
+				h.AssertEq(t, th.Name, "Hives")
+				h.AssertEq(t, th.Typeflag, byte(tar.TypeDir))
+
+				th, _ = tr.Next()
+				h.AssertEq(t, th.Name, "Files/cnb")
+				h.AssertEq(t, th.Typeflag, byte(tar.TypeDir))
+
+				th, _ = tr.Next()
+				h.AssertEq(t, th.Name, "Files/cnb/my-file")
+				h.AssertEq(t, th.Typeflag, byte(tar.TypeReg))
+
+				_, err = tr.Next()
+				h.AssertError(t, err, "EOF")
+			})
+		})
+
+		when("header path begins with slash", func() {
+			it("writes entries", func() {
+				var err error
+
+				f, err := ioutil.TempFile("", "windows-writer.tar")
+				h.AssertNil(t, err)
+				defer func() { f.Close(); os.Remove(f.Name()) }()
+
+				lw := layer.NewWindowsWriter(f)
+
+				h.AssertNil(t, lw.WriteHeader(&tar.Header{
+					Name:     "/cnb/my-file",
+					Typeflag: tar.TypeReg,
+				}))
+
+				h.AssertNil(t, lw.Close())
+
+				f.Seek(0, 0)
+
+				tr := tar.NewReader(f)
+
+				th, _ := tr.Next()
+				h.AssertEq(t, th.Name, "Files")
+				h.AssertEq(t, th.Typeflag, byte(tar.TypeDir))
+
+				th, _ = tr.Next()
+				h.AssertEq(t, th.Name, "Hives")
+				h.AssertEq(t, th.Typeflag, byte(tar.TypeDir))
+
+				th, _ = tr.Next()
+				h.AssertEq(t, th.Name, "Files/cnb")
+				h.AssertEq(t, th.Typeflag, byte(tar.TypeDir))
+
+				th, _ = tr.Next()
+				h.AssertEq(t, th.Name, "Files/cnb/my-file")
+				h.AssertEq(t, th.Typeflag, byte(tar.TypeReg))
+
+				_, err = tr.Next()
+				h.AssertError(t, err, "EOF")
+			})
+		})
+
+		when("header path begins with dot", func() {
```

Comment:
I am not sure relative path inputs even make sense in the context of a layer. We could require file paths to be absolute, or if this just happens to work w/o extra effort, maybe just remove the test so as not to imply that this is a recommended usage.

Reply:
That's a good point. I think we were trying too hard to make it behave like a normal tar writer. But you're right in that this is purely for writing layers, which require absolute paths.